### DISCUSSION TITLE
How to find the min elevation for the moon within 24 hours

### DISCUSSION THREAD

**Discussion by bjarkeandersen** [cosinekitty/astronomy#296] [Q&A] [answered]
Using SearchHourAngle doesn't solve this problem because:

AstroTime astroTime = new(2023, 3, 19, 22, 0, 0);
var observer = new Observer(40.416775, -3.703790, 0);
var minAngle = Astronomy.SearchHourAngle(Body.Moon, observer, 12.0, astroTime);`

gives: minAngle.time = 2023-03-19 22:59:06 , minAngle.hor.altitude = -63.73

AstroTime astroTime = new(2023, 3, 19, 23, 0, 0);
var observer = new Observer(40.416775, -3.703790, 0);
var minAngle = Astronomy.SearchHourAngle(Body.Moon, observer, 12.0, astroTime);

gives: minAngle.time = 2023-03-20 23:51:18 , minAngle.hor.altitude = -57.12

But when time has just passed 2023-03-19 22:59:06 the moons elevation must be very close to -63.73 and therefore lower than -57.12. Any ideas how to solve this problem? What would be the right way to find out the lowest elevation of the moon within 24 hours?

**Comment by cosinekitty** [comment, marked as answer]
Oh yes, that can happen because the Moon takes more than 24 hours to come back to the same hour angle, since it orbits in the same direction the Earth rotates. It is the same reason that the Moon does not always rise (or set) every calendar day.

If you are really wanting to know when the Moon is lowest in a given 24-hour period, even though it is not at a minimum altitude angle, I would do this:

1. Like you do already, search for hour angle = 12 starting at time T1. Let this time be Tx.
2. Also calculate the altitude angle at T1.
3. Let T2 = T1.AddDays(1.0). Then calculate the hour angle at T2.
4. Pick the time T1, T2, or Tx that has the lowest altitude. This will be the time in that 24-hour period that has the smallest altitude angle.

**Reply by cosinekitty** [reply]
I forgot to mention that you have to make sure Tx is between T1 and T2, otherwise you should ignore Tx and pick the minimum altitude from T1 or T2.

Also, I don't know whether this is helpful or not, but SearchHourAngle now accepts a `direction` parameter than can be either `+1` to search forward in time or `-1` to search backward in time. It has a default value of `+1`, but you can pass either value explicitly.

**Comment by bjarkeandersen** [comment]
Thank you for your solution. It works perfectly and was so simple that I'm almost emberrassed I didn't figure that out myself. I must be careful - you answer so quickly that I start asking you instead of using a couple of hours to think for myself ;-)
